### ISSUE TITLE
`MA0039` shouldn't warn when using `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator`

### ISSUE THREAD

**Issue by RenderMichael** [meziantou/Meziantou.Analyzer#648]
### Version of the Meziantou.Analyzer NuGet package

2.0.110

### Rule Identifier

MA0039

### Target Framework

net8.0

### C# Language version

C# 12

### Description

The description for this rule states:

```text
You should not write methods to validate certificates. Those methods are often used to bypass validation altogether, which is bad for security.
```

This is true, however there are times when bypassing validation must intentionally be done. For example, I have a setup involving running integration testing against `localhost`, without SSL validation.

For that reason, .NET exposes [HttpClientHandler.DangerousAcceptAnyServerCertificateValidator](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.dangerousacceptanyservercertificatevalidator).

Use of this property is explicit indication that SSL should be disabled.

Please consider recognizing this property and disabling `MA0039` in it's presence.

### Reproduction Steps

Minimal code:

```c#
using System.Net.Http;

var handler = new HttpClientHandler();

handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
```

**Comment by meziantou** [comment]
> This is true, however there are times when bypassing validation must intentionally be done

In this case you can disable the warning using an attribute or a `#pragma warning disable`

**Comment by RenderMichael** [comment]
That's true, but I feel like using this property conveys a desire to avoid the validation.
